docs/sphinx/dev_guide/ci_tasks.rst

```diff
@@ -12,99 +12,142 @@
+               If the desired entry is not there, but exists in a newer version 
+               of the RADIUSS Spack Configs project, update the RAJA submodule 
+               to use the newer version. If the information does not exist in 
+               any version of the RADIUSS Spack Configs project, create a 
+               branch there, add the needed spec info, and create a pull 
+               request. Then, when that PR is merged, update the RAJA submodule
+               to the new version.
+
+Changing run parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The parameters for each system/system-type on which we run GitLab CI for
+RAJA, such as job time limits, resource allocations, etc. are defined in the 
+``RAJA/.gitlab/custom-jobs-and-variables.yml`` file. This information can
+remain as is, for the most part, and should not be changed unless absolutely 
+necessary.
+
+Sometimes a particular job will take longer to build and run than the 
+default allotted time for jobs on a machine. In this case, the time for the
+job can be adjusted in the job entry in the associated 
+``RAJA/.gitlab/<MACHINE>-build-and-test-extra.yml`` file.
 For example:
 
 .. code-block:: bash
 
-  xl_16_1_1_7_cuda:
-    variables:
-      SPEC: "+cuda %xl@16.1.1.7 cuda_arch=70 ^cuda@10.1.168 ^cmake@3.14.5"
-      DEFAULT_TIME: 60
-    allow_failure: true
-    extends: .build_and_test_on_lassen
-
-This example explicitly sets the build and test allocation time to 60 minutes:
-``DEFAULT_TIME: 60``. Note that it also allows the pipeline to fail: 
-``allow_failure: true``. We do this in some cases where certain tests are known
-to fail regularly. This allows the overall check status to report as passing,
-even though the test pipeline annotated this way may fail.
+  gcc_8_1_0:
+  variables:
+    SPEC: " ${PROJECT_RUBY_VARIANTS} %gcc@8.1.0 ${PROJECT_RUBY_DEPS}"
+    RUBY_BUILD_AND_TEST_JOB_ALLOC: "--time=60 --nodes=1"
+  extends: .build_and_test_on_ruby
 
+This example sets the build and test allocation time to 60 minutes and to 
+run on one node. 
 
-Adding Test Pipelines
----------------------
+Allowing failures
+^^^^^^^^^^^^^^^^^^
 
-Adding a test pipeline involves adding a new entry in the 
-``RAJA/.gitlab-ci.yml`` file.
+Sometimes a shared job configuration is known to fail for RAJA. To allow
+the job to fail without the CI check associated with it failing, we can
+annotate the job for this. For example:
 
-.. important:: Build spec information used in RAJA Gitlab CI pipelines
-               must exist in the ``compilers.yaml`` file and/or 
-               ``packages.yaml`` file for the appropriate system type in
-               the `RADIUSS Spack Configs <https://github.com/LLNL/radiuss-spack-configs>`_ repo.
+.. code-block:: bash
 
+  ibm_clang_9_0_0:
+    variables:
+      SPEC: " ${PROJECT_LASSEN_VARIANTS} %clang@ibm.9.0.0 ${PROJECT_LASSEN_DEPS}"
+    extends: .build_and_test_on_lassen
+    allow_failure: true
 
```

Comment:
Here we should add an "important note" about overriding specs. This was done well in the ci.rst:
Something like:

```suggestion

.. important:: When a shared job needs to be modified for RAJA specifically, we call
             that "overriding": The job label must be kept the same as in the 
             ``<MACHINE>-build-and-test.yml`` file in the 
             `RADIUSS Shared CI <https://github.com/LLNL/radiuss-shared-ci>`_, 
             and the job implementation can be adapted. Please add a comment to
             describe the nature of the change.

```

Reply:
done.